lib/SearchAndSortPane/SearchAndSortPane.js

```diff
@@ -0,0 +1,350 @@
+    nsParams: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.object,
+    ]),
+    path: PropTypes.string,
+    notLoadedMessageId: PropTypes.string,
+    visibleColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+    columnMapping: PropTypes.object.isRequired,
+    columnWidths: PropTypes.object.isRequired,
+    resultsFormatter: PropTypes.object.isRequired,
+    defaultSort: PropTypes.string,
+    resourceName: PropTypes.string.isRequired,
+    resultCountIncrement: PropTypes.number.isRequired,
+    initialResultCount: PropTypes.number.isRequired,
+  };
+
+  static defaultProps = {
+    maxSortKeys: 2,
+    renderHeader: null,
+    defaultSort: '',
+    searchLabelKey: 'stripes-data-transfer-components.search',
+  };
+
+  constructor(props) {
+    super(props);
+
+    const { match: { path: routePath } } = this.props;
+
+    this.state = { changedTerm: undefined };
+    this.initialQuery = queryString.parse(routePath);
+  }
+
+  componentDidMount() {
+    this.setInitialSortQueryParam();
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      stripes,
+      resourceName,
+    } = this.props;
+
+    const logger = stripes ? stripes.logger : { log: noop };
+    const oldState = makeConnectedSource(prevProps, logger, resourceName);
+    const newState = makeConnectedSource(this.props, logger, resourceName);
+
+    const isSearchComplete = oldState.pending() && !newState.pending();
+
+    if (isSearchComplete) {
+      const count = newState.totalCount();
+
+      this.SRStatusRef.current.sendMessage((
+        <FormattedMessage
+          id="stripes-smart-components.searchReturnedResults"
+          values={{ count }}
+        />
+      ));
+    }
+  }
+
+  SRStatusRef = createRef();
+
+  setInitialSortQueryParam() {
+    const {
+      defaultSort,
+      location: { search },
+    } = this.props;
+
+    const queryParams = queryString.parse(search);
+    const sortOrder = queryParams.sort || defaultSort;
+
+    this.transitionToParams({ sort: sortOrder });
+  }
+
+  transitionToParams(values) {
+    const {
+      nsParams,
+      location,
+      history,
+    } = this.props;
+
+    const nsValues = mapNsKeys(values, nsParams);
+    const url = buildUrl(location, nsValues);
+
+    history.push(url);
+  }
+
+  getSource() {
+    const {
+      stripes,
+      resourceName,
+    } = this.props;
+
+    return makeConnectedSource(this.props, stripes ? stripes.logger : { log: noop }, resourceName);
+  }
+
+  onSort = (e, meta) => {
+    const {
+      maxSortKeys,
+      defaultSort,
+    } = this.props;
+
+    const sortOrder = buildSortOrder(this.queryParam('sort') || defaultSort, meta.name, defaultSort, maxSortKeys);
+
+    this.transitionToParams({ sort: sortOrder });
+  };
+
+  queryParam(name) {
+    const {
+      parentResources: { query },
+      nsParams,
+    } = this.props;
+
+    const nsKey = getNsKey(name, nsParams);
+
+    return get(query, nsKey);
+  }
+
+  onSearchChange = e => {
```

Comment:
Let's come up with `handle` instead of `on` e.g. `handleSearchChange`. The same for `onSort`, `onSearchSubmit` and `onSearchQueryClear`.